okhttp/src/main/java/io/grpc/okhttp/OkHttpClientStream.java

```diff
@@ -318,6 +318,12 @@ public void transportDataReceived(okio.Buffer frame, boolean endOfStream) {
       super.transportDataReceived(new OkHttpReadableBuffer(frame), endOfStream);
     }
 
+    // Safe to call without lock
+    @SuppressWarnings("GuardedBy")
+    public Attributes getAttributes() {
+      return transport.getAttributes();
+    }
```

Comment:
You can get the attributes and save it as a field at `TransportState` constructor, the transport is guaranteed in READY state when the constructor is called.

Reply:
Done. This required updating some unit tests to ensure streams are not started until the transport is ready.